packages/client/lib/CredentialOffer/IssuanceInitiation.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  EndpointMetadata,
+  IssuanceInitiationRequestPayload,
+  IssuanceInitiationWithBaseUrl,
+  OIDCVCIVersion
+} from '@sphereon/openid4vci-common';
+import Debug from 'debug';
+
+import {MetadataClient} from "../MetadataClient";
+import { convertJsonToURI, convertURIToJsonObject } from '../functions';
+
+import {CredentialOfferStrategy} from "./index";
+
+const debug = Debug('sphereon:openid4vci:initiation');
+export class IssuanceInitiation implements CredentialOfferStrategy {
+  readonly version: OIDCVCIVersion;
+  private readonly _issuanceInitiationWithBaseUrl: IssuanceInitiationWithBaseUrl;
+
+  public constructor(issuanceInitiationURI: string){
+    this.version = OIDCVCIVersion.VER_9;
+    this._issuanceInitiationWithBaseUrl = this.fromURI(issuanceInitiationURI);
+  }
+
+  public getPayload() : IssuanceInitiationWithBaseUrl {
+    return this._issuanceInitiationWithBaseUrl;
+  }
+
+  private fromURI(issuanceInitiationURI: string): IssuanceInitiationWithBaseUrl {
```

Comment:
I would expect this to be a public static preparing everything and then creating the object through the constructor instead of doing it the other way around. This also would allow us to go async if we need it in the future

Reply:
fixed.